javascript/javascript1/week3/script1.js

```diff
@@ -36,9 +36,78 @@ function notThisFunctionName(travelInformation) {
     console.log(hours);
     let minutes = Math.floor((time - hours) * 60);
     return hours + " Hours " + minutes + " Minutes";
-
 }
 
 const travelTime = notThisFunctionName(travelInformation);
 
 console.log(travelTime); // 8 hours and 38 minutes
+
+
+//Exe-3 Series duration of my life:
+const seriesDurations = [
+    {
+        title: "Game of thrones",
+        days: 3,
+        hours: 1,
+        minutes: 0,
+    },
+    {
+        title: "Sopranos",
+        days: 3,
+        hours: 14,
+        minutes: 0,
+    },
+    {
+        title: "The Wire",
+        days: 2,
+        hours: 12,
+        minutes: 0,
+    },
+];
+// Adding one more favorite TV show to the array.
+let myShow = {
+    title: "Emily in paris",
+    days: 6,
+    hours: 5,
+    minutes: 0,
+};
+seriesDurations.push(myShow);
+console.log(seriesDurations);
+// write function to calculate how much time you spent
+
+const totalTimeInMinutes0 = (seriesDurations[0].days * 24 * 60) + (seriesDurations[0].hours * 60) + seriesDurations[0].minutes;
+const totalTimeInMinutes1 = (seriesDurations[1].days * 24 * 60) + (seriesDurations[1].hours * 60) + seriesDurations[1].minutes;
+const totalTimeInMinutes2 = (seriesDurations[2].days * 24 * 60) + (seriesDurations[2].hours * 60) + seriesDurations[2].minutes;
+const totalTimeInMinutes3 = (seriesDurations[3].days * 24 * 60) + (seriesDurations[3].hours * 60) + seriesDurations[3].minutes;
+const totalLifeSpanInMinutes = 80 * 365 * 24 * 60;
+
+function logOutSeriesText(title) {
+    // write code here
+    let totalLifeSpent0 = ((totalTimeInMinutes0 * 100) / totalLifeSpanInMinutes).toFixed(3);
+    let totalLifeSpent1 = ((totalTimeInMinutes1 * 100) / totalLifeSpanInMinutes).toFixed(3);
+    let totalLifeSpent2 = ((totalTimeInMinutes2 * 100) / totalLifeSpanInMinutes).toFixed(3);
+    let totalLifeSpent3 = ((totalTimeInMinutes3 * 100) / totalLifeSpanInMinutes).toFixed(3);
+
+    if (title == seriesDurations[0].title) {
+        console.log(seriesDurations[0].title + " took " + totalLifeSpent0 + "% of my life");
+    } else if (title == seriesDurations[1].title) {
+        console.log(seriesDurations[1].title + " took " + totalLifeSpent1 + "% of my life");
+    } else if (title == seriesDurations[2].title) {
+        console.log(seriesDurations[2].title + " took " + totalLifeSpent2 + "% of my life");
+    } else if (title == seriesDurations[3].title) {
+        console.log(seriesDurations[3].title + " took " + totalLifeSpent3 + "% of my life");
```

Comment:
I think you can combine all of this in a for loop. I did something like
`let sumPercentage = 0;
	for (let i = 0; i < seriesDurations.length; i++) {
		const timeInMinutes =
			seriesDurations[i].days * 24 * 60 +
			seriesDurations[i].hours * 60 +
			seriesDurations[i].minutes * 60;
		const percentage = (timeInMinutes / (80 * 525600)) * 100;
		console.log(
			`${seriesDurations[i].title} took ${percentage.toFixed(4)} of my life`
		);
		sumPercentage += percentage;
	}`

Reply:
Yes i tried this one but still have some doubts. may be we can discuss that in class.

Reply:
I would agree with Andrii.

Reply:
Yes i made a loop and its works now thanks both of you